ethpm/backends/http.py

```diff
@@ -1,37 +1,44 @@
-from urllib import parse
+import base64
+import json
 
 import requests
 
 from ethpm.backends.base import BaseURIBackend
-from ethpm.constants import RAW_GITHUB_AUTHORITY
-from ethpm.utils.uri import is_valid_github_uri
-from ethpm.validation import validate_uri_contents
+from ethpm.constants import GITHUB_API_AUTHORITY
+from ethpm.exceptions import CannotHandleURI
+from ethpm.typing import URI
+from ethpm.utils.uri import is_valid_content_addressed_github_uri
 
 
 class GithubOverHTTPSBackend(BaseURIBackend):
     """
     Base class for all URIs pointing to a content-addressed Github URI.
     """
 
-    def can_resolve_uri(self, uri: str) -> bool:
-        return is_valid_github_uri(uri)
+    def can_resolve_uri(self, uri: URI) -> bool:
+        return is_valid_content_addressed_github_uri(uri)
 
-    def can_translate_uri(self, uri: str) -> bool:
+    def can_translate_uri(self, uri: URI) -> bool:
         """
         GithubOverHTTPSBackend uri's must resolve to a valid manifest,
         and cannot translate to another content-addressed URI.
         """
         return False
 
-    def fetch_uri_contents(self, uri: str) -> bytes:
-        parsed_uri = parse.urlparse(uri)
-        validation_hash = parsed_uri.fragment
-        http_uri = f"{parsed_uri.scheme}://{parsed_uri.netloc}{parsed_uri.path}"
-        response = requests.get(http_uri)
+    def fetch_uri_contents(self, uri: URI) -> bytes:
+        if not self.can_resolve_uri(uri):
+            raise CannotHandleURI(f"GithubOverHTTPSBackend cannot resolve {uri}.")
+
+        response = requests.get(uri)
         response.raise_for_status()
-        validate_uri_contents(response.content, validation_hash)
-        return response.content
+        contents = json.loads(response.content)
+        if contents["encoding"] != "base64":
+            raise CannotHandleURI(
+                "Expected contents returned from Github to be base64 encoded, "
+                f"instead received {contents['encoding']}."
+            )
+        return base64.b64decode(contents["content"])
```

Comment:
Maybe I missed it but I expected to see somewhere that we verify that the contents match the hash.  If that isn't present I think it needs to be added.

Reply:
Yup, I chose to remove that for some reason that escapes me now, I'll set about re-implementing that.